### ISSUE TITLE
JSON-LD Writer Does Not Write @type when Subject is owl:Class

### ISSUE THREAD

**Issue by daltontc** [eclipse/rdf4j#2707]
I have a use case where I use an ontology with the triple `<http://www.w3.org/2002/07/owl#Class> a <http://www.w3.org/2002/07/owl#Class> .`. Starting in RDF4j 2.5.0 through to 3.x the JSON-LD writer does not properly convert this triple. It fully spells out the type predicate instead of using "@type".

```
[ {
  "@id" : "http://www.w3.org/2002/07/owl#Class",
  "http://www.w3.org/1999/02/22-rdf-syntax-ns#type" : [ {
    "@id" : "http://www.w3.org/2002/07/owl#Class"
  } ]
} ]
```

I created a sample repo to demonstrate this: https://github.com/daltontc/rdf4jTest

If the runtime version is set to 2.4.6, the JSON-LD is formatted as expected.

**Comment by barthanssens** [comment]
Hmz, I'll have to look into this, probably related to an upgrade of the JSON-LD library: RDF4J 2.4.6 uses jsonld-java 0.12.0, while 2.5 uses 0.12.3...

**Comment by barthanssens** [comment]
Same in RDF4J 3.5, setting USE_RDF_TYPE to true or false does not make a difference, and not limited to owl:Class (seems like generic issue when subject and object are the same)

**Comment by barthanssens** [comment]
It's indeed a jsonld-java issue...

**Comment by daltontc** [comment]
Any chance the fix will be available for 2.x? Or will it only be a 3.x fix?

**Comment by barthanssens** [comment]
Well, if you need a solution quickly, you may want to exclude the jsonld-java 0.13 from your project pom.xml, and include the older jsonld-java 0.12.2 instead... That should do the trick...

If it gets fixed in jsonld-java, it will most likely be a rdf4j 3.x -only fix.

Is there a specific reason why you'd prefer a 2.x version as well ? 
For most projects, 3.x provides many enhancements and useful features (and in addition, 3.x is tested with JDK 11+), so in general we don't release 2.x versions anymore, unless it's really really needed...

**Comment by barthanssens** [comment]
Fixed in the upcoming jsonld-java v0.13.3

**Comment by barthanssens** [comment]
@daltontc fixed in the upcoming 3.7.0 release

In the mean time, explicitly adding the recently released jsonld-java v 0.13.3 dependency to the pom / gradle file might do the trick
